src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1929,12 +1929,19 @@ private CSDiagnosticInfo NotFound(SyntaxNode where, string simpleName, int arity
                         fullName = qualifierOpt.ToDisplayString(SymbolDisplayFormat.QualifiedNameOnlyFormat) + "." + fullName;
                     }
 
-                    forwardedToAssembly = GetForwardedToAssembly(fullName, arity, out encounteredForwardingCycle);
+                    forwardedToAssembly = GetForwardedToAssembly(fullName, arity, out encounteredForwardingError);
 
-                    if (encounteredForwardingCycle)
+                    if (encounteredForwardingError != null)
                     {
-                        Debug.Assert((object)forwardedToAssembly != null, "How did we find a cycle if there was no forwarding?");
-                        diagnostics.Add(ErrorCode.ERR_CycleInTypeForwarder, location, fullName, forwardedToAssembly.Name);
+                        if (encounteredForwardingError.Code == (int)ErrorCode.ERR_CycleInTypeForwarder)
+                        {
+                            Debug.Assert((object)forwardedToAssembly != null, "How did we find a cycle if there was no forwarding?");
+                            diagnostics.Add(ErrorCode.ERR_CycleInTypeForwarder, location, fullName, forwardedToAssembly.Name);
+                        }
+                        else if (encounteredForwardingError.Code == (int)ErrorCode.ERR_TypeForwardedToMultipleAssemblies)
+                        {
+                            return diagnostics.Add(ErrorCode.ERR_TypeForwardedToMultipleAssemblies, location, encounteredForwardingError.Arguments);
+                        }
                     }
```

Comment:
These are the only two callers to `GetForwardedToAssembly`. Could the `DiagnosticBag` and `location` be passed into that method and create the diagnostic there?